File: atividade-3/modulo-4/007_somaImpares/007_somaImpares.js
//Defina uma função somaImpares tal que somaImpares xs devolve a soma dos elementos ímpares de uma lista.

const list = [2, 5, 4, 7, 9, 6];

const somaImpares = (xs) =>
  xs.filter((x) => x % 2 !== 0).reduce((acc, current) => acc + current);

console.log(somaImpares(list));

File: atividade-3/modulo-4/007_somaImpares/007_somaImpares.hs
module Main where

-- Defina uma função somaImpares tal que somaImpares xs devolve a soma dos elementos ímpares de uma lista.

somaImpares :: [Int] -> Int
somaImpares xs = foldl (\x y -> x + y) 0 [x | x <- xs, odd x]

main = do
  a <- readLn :: IO [Int]
  print $ somaImpares a
